warp10/src/main/java/io/warp10/continuum/gts/GTSDecoder.java

```diff
@@ -528,26 +528,52 @@ public Object getBinaryValue() {
   /**
    * Decode any remaining values into a GTS instance.
    * 
+   * @param type TYPE to force for the resulting GTS
+   * @param strict Set to true to force values to be of uniform types, will throw RuntimeException if not
+   * 
    * @return A GTS instance containing the remaining values.
    */
-  public GeoTimeSerie decode(TYPE type) {
+  public GeoTimeSerie decode(TYPE type, boolean strict) {
     GeoTimeSerie gts = new GeoTimeSerie(this.count > 0 ? (int) Math.min(Integer.MAX_VALUE, this.count) : Math.max(16, this.buffer.remaining() / 10));
     
     if (null != type) {
       gts.setType(type);
     }
     
     gts.setMetadata(this.getMetadata());
-    
-    while(next()) {
-      GTSHelper.setValue(gts, getTimestamp(), getLocation(), getElevation(), getValue(), false);
+  
+    if (strict) {
+      Class lastClass = null;
+      
+      while(next()) {        
+        Object value = getValue();
+        Class valClass = value.getClass();
+        // getValue could return a BigDecimal, we need to smooth the comparison so
+        // they are considered as Double
+        if (value instanceof BigDecimal) {
+          valClass = Double.class;
+        }
+        if (null != lastClass && !(valClass.equals(lastClass))) {
+          throw new RuntimeException("Non homogeneous GTS Encoder.");
```

Comment:
Creating and using a subclass of RuntimeException could be nicer.

Reply:
The only reason it's a RuntimeException is that it avoids declaring it is thrown, no plan to subclass it.